### DISCUSSION TITLE
nxt editor causes rig to collapse and crashes

### DISCUSSION THREAD

**Discussion by geordiemartinez** [nxt-dev/nxt_editor#167] [General]
Here are 2 vidcaps of various issues.   It's a standard Advanced Skeleton biped. no plugins required.  
I suspect there is something with the logger that is causing issues. 

this first vidcap shows the rig loading doesn't finish because of some error that is captured by nxt so the .ma rig never fully gets finished loading. 
 https://youtu.be/Oa94y90pWkc

This second video is 20 min of testing various logger settings on the nxt editor to see which ones are snagging errors and crashing maya, 
https://youtu.be/Tof492IIUnI

I will provide the rig file upon request, but can't paste it here.

**Comment by MichaelAldrich** [comment]
👋 Hi @geordiemartinez, I did a quick test with one of the example Advanced Skeleton rigs, and I was not able to repro the issue specifically. If you can share the rig with dev@opennxt.dev we can see what testing we can do. Thank you for the detailed testing videos, they will make it easier to find your issue.

**Reply by geordiemartinez** [reply]
Thank you. I have shared the asset via email

**Reply by MichaelAldrich** [reply]
Good news! I was able to reproduce the frequent instability of loading your rig. I expect the problem is due to the logging callback [we install into maya here](https://github.com/nxt-dev/nxt_editor/blob/release/nxt_editor/integration/maya/plug-ins/nxt_maya.py#L133).

I expect what happens is that the sheer volume of `NAN` errors that are output overwhelm our logging functions. As a test measure, can you try launching nxt via the following python in the maya script editor?
```
import nxt_editor
nxt_editor.show_new_editor()
```

This will launch a less integrated version of the nxt editor inside of maya, and will not install the aformentioned callbacks. In my testing when launching the editor this way I was able to open the rig several times with no problems.

The maya plugin is due for a rebuild based on the [learnings we're doing from expanding our DCC support](https://github.com/nxt-dev/nxt/issues/65). If this test works for you we can investigate putting together a proper fix in the next release.

**Comment by geordiemartinez** [comment]
Thank you Michael! I will test this out.
